### PR TITLE
update workspace resourceType to support catalog action and ADs

### PR DESCRIPTION
**Pull request by MatthewBemis**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

**Review by dvoet — APPROVED** [review]

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -58,7 +61,11 @@ resourceTypes = {
       can-compute = {
         roleActions = ["compute"]
       }
+      can-catalog = {
+        roleActions = ["catalog"]
+      }
     }
+    authDomainConstrainable = true
```
Comment (Qi77Qi) [inline src/main/resources/reference.conf:68]: does this need to be added to test's reference.conf as well? 🤔

Comment (MatthewBemis) [reply src/main/resources/reference.conf:68]: no i don't think so. why do you ask?

Comment (Qi77Qi) [reply src/main/resources/reference.conf:68]: nothing specific, just wonder if it should be added in test's reference.conf and possibly be tested

Comment (MatthewBemis) [reply src/main/resources/reference.conf:68]: it's already in there for some test-specific resource type

**Pull request merged**